### PR TITLE
Allow running image build commands in order

### PR DESCRIPTION
**Pull request by nickpetrovic**
Adds new `add_commands()` and `add_python_packages()` to the `Image` abstraction.

This allows for running commands or installing python packages in order, but only after the initial values of the Image's attributes `commands` and `python_packages`. The reason for the exception is to maintain backwards compatibility with existing stubs/deployments.

Resolve BE-1675

### CODE REVIEW

File: pkg/abstractions/image/build.go [hunk 1]
```diff
@@ -37,6 +40,11 @@ type Builder struct {
 	tailscale     *network.Tailscale
 }
 
+type ContainerCommand struct {
```
Comment (luke-lombardi) [inline pkg/abstractions/image/build.go:43]: If we end up supporting things like `with_cuda_version` or something similar (things that aren't necessarily a container command), is this the right naming? I was thinking maybe something along the lines of `ContainerBuildStep` (or something like that)

Users would need to know either way, but for internal clarity

**Review by luke-lombardi — CHANGES_REQUESTED** [review]
this looks good, just had a concern around naming

File: pkg/abstractions/image/build.go (hunk 1 above)
Comment (nickpetrovic) [reply pkg/abstractions/image/build.go:43]: Yeah good point. I'll change it.

Comment (nickpetrovic) [reply pkg/abstractions/image/build.go]: Updated to `BuildStep`.

**Review by luke-lombardi — APPROVED** [review]

**Pull request merged**